### PR TITLE
chore(core): check blob length when verifying metadata

### LINKED ISSUE #187: When verifying the metadata, also check the `unencoded_length`
Currently the `verify` function on `UnverifiedBlobMetadataWithId ` does not check that the metadata has a valid size.

### PR DESCRIPTION
**Pull request by giac-mysten**
Closes #187
Depends on #194

### CODE REVIEW

File: crates/walrus-core/src/metadata.rs [hunk 1]
```diff
@@ -159,17 +160,31 @@ impl UnverifiedBlobMetadataWithId {
...
                 actual: n_hashes,
-                expected: n_shards.get(),
+                expected: config.n_shards as usize,
             }
         );
 
+        // Check that the `unencoded` length does not exceed what can be encoded with the number of
+        // symbols given.
+        crate::ensure!(
+            config
```
Comment (gdanezis) [inline crates/walrus-core/src/metadata.rs]: My preference: give a descriptive name to the return of the call, then put the condition including the descriptive name in the ensure, to make all this more readable. Variable names are natural comments.

Comment (giac-mysten) [reply crates/walrus-core/src/metadata.rs]: makes sense! will change this

File: crates/walrus-core/src/metadata.rs [hunk 2]
```diff
@@ -159,23 +160,31 @@ impl UnverifiedBlobMetadataWithId {
...
+        let symbol_size_from_metadata = config.symbol_size_for_blob(
+            self.metadata
+                .unencoded_length
+                .try_into()
+                .map_err(|_| VerificationError::UnencodedLengthTooLarge)?,
+        );
+        crate::ensure!(
+            symbol_size_from_metadata.is_some(),
+            VerificationError::UnencodedLengthTooLarge
+        );
```
Comment (mlegner) [inline crates/walrus-core/src/metadata.rs]: Remark: This feels a bit convoluted to me. How about this version?
```suggestion
        crate::ensure!(
            self.metadata.unencoded_length <= config.max_blob_size() as u64,
            VerificationError::UnencodedLengthTooLarge
        );
```

File: crates/walrus-service/src/client/communication.rs [hunk 3]
```diff
@@ -152,8 +152,17 @@ impl<'a> NodeCommunication<'a> {
...
-                        .symbol_size_for_blob(metadata.metadata().unencoded_length.try_into()?)
-                        .ok_or(anyhow!("the blob size in the metadata is too large"))?,
+                        .symbol_size_for_blob(
+                            metadata
+                                .metadata()
+                                .unencoded_length
+                                .try_into()
+                                .expect("checked in `UnverifiedBlobMetadataWithId::verify`")
+                        )
+                        .ok_or(anyhow!("the blob size in the metadata is too large"))
```
Comment (mlegner) [inline crates/walrus-service/src/client/communication.rs]: Nit: This line can be removed.
```suggestion
```

**Pull request merged**